runtime/interpreter/decode.go

```diff
@@ -350,6 +350,8 @@ func (d StorableDecoder) decodeInt() (IntValue, error) {
 }
 
 func (d StorableDecoder) decodeInt8() (Int8Value, error) {
+	common.UseMemory(d.memoryGauge, int8MemoryUsage)
+
 	v, err := d.decoder.DecodeInt64()
```

Comment:
As a `int64` is decoded, it should be metered as such. Maybe define a function `decodeInt64` which meters 8 bytes and calls the `StreamDecoder.DecodeInt64`, e.g. like the `decodeString` helper function

Reply:
added with e76c1f6